d2/src/main/java/com/linkedin/d2/balancer/config/DarkClustersConverter.java

```diff
@@ -49,36 +59,115 @@
     }
     else
     {
-      try
-      {
-        String json = CODEC.mapToString(config.data());
-        return JacksonUtil.getObjectMapper().readValue(json, Map.class);
```

Comment:
Can we keep this old behavior of using Jackson to serialize everything and selectively replace each transport properties with output of TransportClientPropertiesConverter?

Reply:
that seems pretty hacky. Doing what I did (use the Map<String, Object>) is more consistent with what the code was doing for other stuff, anyway. The only reason I was trying to use the pegasus object for both in-mem and for zookeeper storage was because I didn't see the need to have two formats; with the inclusion of transportClientProperties I do need the separate format. Regarding your proposal, that would require retraversing the DarkClusterConfigMap and accessing the dataMap directly to hardcode some mappings, and wouldn't take into account further changes in the TransportClientPropertiesConverter. So I don't think that's a better option. Doing the Map<String, Object> way may be more code, but it's pretty understandable and in line with the rest of the Converter classes.

Reply:
ok